hathor/verification/verification_coordinator.py

```diff
@@ -0,0 +1,231 @@
+                self._tx_storage.get_vertex(dep_id)
+            except TransactionDoesNotExist:
+                missing_deps.append(dep_id)
+
+        return missing_deps
+
+    def _get_verification_model(self, vertex: Vertex) -> VerificationModel:
+        """
+        Return the verification model for a vertex using dependencies that may or may not have been persisted yet.
+        """
+        # We assert with type() instead of isinstance() because each subclass has a specific branch.
+        match vertex.version:
+            case TxVersion.REGULAR_BLOCK:
+                assert type(vertex) is Block
+                basic_deps, deps = self._get_block_deps(vertex)
+                return VerificationBlock(
+                    vertex=vertex,
+                    basic_deps=basic_deps,
+                    deps=deps,
+                )
+
+            case TxVersion.MERGE_MINED_BLOCK:
+                assert type(vertex) is MergeMinedBlock
+                basic_deps, deps = self._get_block_deps(vertex)
+                return VerificationMergeMinedBlock(
+                    vertex=vertex,
+                    basic_deps=basic_deps,
+                    deps=deps,
+                )
+
+            case TxVersion.REGULAR_TRANSACTION:
+                assert type(vertex) is Transaction
+                return VerificationTransaction(
+                    vertex=vertex,
+                    basic_deps=None,
+                    deps=self._get_tx_deps(vertex),
+                )
+
+            case TxVersion.TOKEN_CREATION_TRANSACTION:
+                assert type(vertex) is TokenCreationTransaction
+                return VerificationTokenCreationTransaction(
+                    vertex=vertex,
+                    basic_deps=None,
+                    deps=self._get_tx_deps(vertex),
+                )
+
+            case _:
+                assert_never(vertex.version)
+
+    def _get_block_deps(self, block: Block) -> tuple[BasicBlockDependencies, BlockDependencies]:
+        """
+        Return block dependencies for a block using dependencies that may or may not have been persisted yet.
+        """
+        parents = {vertex_id: self._get_vertex(vertex_id) for vertex_id in block.parents}
+        daa_deps = self._daa.get_block_dependencies(block, self._get_parent_block)
+
+        basic_deps = BasicBlockDependencies(
+            parents=parents,
+            daa_deps=daa_deps,
+        )
+
+        deps = BlockDependencies(parents=parents)
+
+        return basic_deps, deps
+
+    def _get_tx_deps(self, tx: Transaction) -> TransactionDependencies:
+        """
+        Return transaction dependencies for a transaction using dependencies that may or may not have been persisted
+        yet.
+        """
+        parents = {vertex_id: self._get_vertex(vertex_id) for vertex_id in tx.parents}
+        spent_txs = {tx_input.tx_id: self._get_vertex(tx_input.tx_id) for tx_input in tx.inputs}
+        token_info = tx.get_complete_token_info()
```

Comment:
I just noticed this will fail for transactions because this method accesses the storage. We need to make it accept functional dependencies like the other methods.